codegen/chainconfig.go

```diff
@@ -1,143 +1,47 @@
 package codegen
 
-type ChainConfig struct {
-	ID               string // Public
-	DisplayName      string // Public
-	ExplorerLink     string
-	ApiEndpoint      string
-	FirehoseEndpoint string
-	Network          string
-	SupportsCalls    bool
+import (
+	"log"
+	"sync"
+
+	registry "github.com/pinax-network/graph-networks-libs/packages/golang/lib"
+)
+
+var (
+	registryNetworks     map[string]*registry.Network
+	registryNetworksOnce sync.Once
+)
+
+// GetRegistryNetworks returns a map of network ID to *registry.Network, using Pinax's registry as primary source.
+// Only networks with a Firehose endpoint are included.
+// If fetching fails, it falls back to loading from a local JSON file ("TheGraphNetworksRegistry.json").
+func GetRegistryNetworks() map[string]*registry.Network {
+	registryNetworksOnce.Do(func() {
+		reg, err := registry.FromLatestVersion()
+		if err != nil {
+			// Fallback: try to load from local file
+			// TODO: Validate where to put the registry file
+			reg, err = registry.FromFile("TheGraphNetworksRegistry.json")
+			if err != nil {
+				log.Fatalf("Failed to load registry from both network and file: %v", err)
+			}
+		}
+		m := make(map[string]*registry.Network)
+		for i, net := range reg.Networks {
+			if len(net.Services.Firehose) > 0 {
+				m[net.ID] = &reg.Networks[i]
+			}
+		}
+		registryNetworks = m
+		// Optionally add custom networks for testing
+		addCustomNetwork()
+	})
+	return registryNetworks
 }
 
-var ChainConfigByID = map[string]*ChainConfig{
-	"mainnet": {
-		DisplayName:      "Ethereum Mainnet",
-		ExplorerLink:     "https://etherscan.io",
-		FirehoseEndpoint: "mainnet.eth.streamingfast.io:443",
```

Comment:
Could you take all keys previously defined here and create a unit tests that ensures that all of them can be retrieved today so we ensure that code:

```
networks := GetRegistryNetworks()
	net := networks[p.Network]
```

Works for all previous cases.

Reply:
test fails becuause some of the `id` are different:
```
bnb -> bsc
polygon -> matic
amoy -> polygon-amoy
arbitrum -> arbitrum-one
```

But if we use the `Find` method it works. I guess the test is really to make sure all the stuff is there, so that's fine.